client/app/components/Chart/__tests__/ChartControl.spec.jsx

```diff
@@ -7,28 +7,79 @@ import React from 'react';
 import { ChartControl } from '../ChartControl';
 
 describe('ChartControl', () => {
+  // mocked data
+  const MOMENTS_TYPE = 'Moments';
+  const CATEGORIES_TYPE = 'Categories';
+  const MOODS_TYPE = 'Moods';
+  const types = [MOMENTS_TYPE, CATEGORIES_TYPE, MOODS_TYPE];
+  const data = {
+    [MOMENTS_TYPE]: {
+      '2013-02-10 00:00:00 -0800': 11,
+      '2013-02-11 00:00:00 -0800': 6,
+    },
+    [CATEGORIES_TYPE]: [
+      {
+        data: {
+          '2013-02-10': 3,
+          '2013-02-11': 6,
+          '2013-02-12': 13,
+        },
+        name: 'School',
+      },
+    ],
+    [MOODS_TYPE]: [
+      {
+        data: {
+          '2013-02-10': 2,
+          '2013-02-11': 8,
+          '2013-02-12': 3,
+        },
+        name: 'Anxious',
+      },
+    ],
+  };
+
   it('renders', () => {
     let wrapper = null;
     expect(() => {
       // shallow render only lists the first level of components
       wrapper = shallow(
         <ChartControl
-          types={['Moments']}
-          initialParams={{
-            type: 'Moments',
-            data: {
-              Moments: {
-                '2013-02-10 00:00:00 -0800': 11,
-                '2013-02-11 00:00:00 -0800': 6,
-              },
-            },
-          }}
+          types={types}
+          initialParams={{ type: MOMENTS_TYPE, data }}
         />,
       );
     }).not.toThrow();
 
     expect(wrapper).not.toBeNull();
-    expect(wrapper.find('ChartControlButton').length).toEqual(1);
+    expect(wrapper.find('ChartControlButton').length).toEqual(types.length);
     expect(wrapper.find('Chart').length).toEqual(1);
   });
+
+  it('passes the expected data for the selected type', () => {
+    const wrapper = shallow(
```

Comment:
So we actually have another [issue](https://github.com/ifmeorg/ifme/issues/1789) to use React Testing Library instead of Enzyme because we are moving away from it. Would you be interested in tackling this too? :)

Reply:
Yes absolutely! I was just wondering about it when adding the new test. I can take that one on.

Reply:
Yes feel free to take it on!

Reply:
You can do it in another PR if you want!

Reply:
Awesome! Will do